### DISCUSSION TITLE
paste on cursor position in insert mode

### DISCUSSION THREAD

**Discussion by ghost** [neovide/neovide#2099] [General]
sometimes i like to paste without leaving insert mode. in nvim that can be tricky, because afaik any keybind that can do that relies on either < C-o > or < Esc >, which both leave insert mode (temporarily) and therefore change the cursor position.

this leads to problems when pasting at the very beginning or end of a line. with < C-o >P, if the cursor is at the very end of the line, it will paste the clipboard one character to the left (like -- content-of-linPASTEe -- instead of -- content-of-linePASTE)

with < Esc > theres the same problem, but at the very beginning of the line. i tried writing a lua function to solve this problem, but it seems impossible, because of the cursor changing position caused by (temporarily) leaving insert mode. no matter how i write that function, the paste will be messed up either at the beginning or the end of the line.

if i use nvim in my terminal (alacritty) this can be solved with ctrl+shift+v. this will paste the clipboard exactly where the cursor is in insert mode, even at the very beginning/end of the line. unfortunately that feature is not available in neovide and i wasnt able to find any solution or workaround for this

**Comment by fredizzimo** [comment]
https://neovim.io/doc/user/insert.html#i_CTRL-R

**Reply by ghost** [reply]
thank you! < C-r > < C-o > " does exactly what i want
